### PR TITLE
Censor Window filter

### PR DESCRIPTION
**Pull request by pavgra**
For https://github.com/OHDSI/circe-be/issues/26

### CODE REVIEW

File: src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java [hunk 1]
```diff
@@ -303,6 +303,32 @@ public String getCollapseConstructorQuery(CollapseSettings collapseSettings) {
...
+    if (censorWindow != null && (censorWindow.startDate != null || censorWindow.endDate != null)) {
+      if (censorWindow.startDate != null) {
+        String censorStartDate = dateStringToSql(censorWindow.startDate);
+        startDate = "CASE WHEN start_date > " + censorStartDate + " THEN start_date ELSE " + censorStartDate + " END";
+      }
+      if (censorWindow.endDate != null) {
+        String censorEndDate = dateStringToSql(censorWindow.endDate);
+        endDate = "CASE WHEN end_date < " + censorEndDate + " THEN end_date ELSE " + censorEndDate + " END";
+      }
+      query += "\nWHERE @start_date <= @end_date";
```
Comment (chrisknoll) [inline src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java]: This is an elegant way to drop people who aren't bounded by the study window, nice!

File: pom.xml [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
     <groupId>org.ohdsi</groupId>
     <artifactId>circe</artifactId>
-    <version>1.3.1-SNAPSHOT</version>
+    <version>1.3.2-SNAPSHOT</version>
```
Comment (chrisknoll) [inline pom.xml]: Per semver semantics, I think this should be updated to 1.4.0-SNAPSHOT:  we aren't doing a bugfix, this is a new feature which doesn't break existing functionality (at least based on this: https://semver.org/)

If you agree, can you update the version to 1.4.0-SNAPSHOT?

**Comment by pavgra** [comment]
@chrisknoll, version was fixed

**Comment by chrisknoll** [comment]
Thanks @pavgra . before making an official relase of 1.4.0, I'm going to let a little time pass to catch any last minute changes that anyone might want to contribute.  I'll schedule a 1.4.0 release for 2/8.

**Pull request merged**